src/pudl/transform/ferc1.py

```diff
@@ -249,6 +250,49 @@ def align_row_numbers_dbf(
     return df
 
 
+class SelectDateRangeDurationXbrl(TransformParams):
```

Comment:
does this name actually make sense? the "range"-ness is what is tripping me up. This transformer selects *specific* start and end dates... which is kind of like a range but if the dates in the table were *inside* of the given range rather than the specific range given, it would not return those (i.e. if you gave it a start-of-year start and end-of-year end but there were some middle quarter data in the df, the quarterly data would not be included.)

Reply:
I agree this doesn't quite seem right. How much flexibility do we actually want this to have? It seems like right now what we need it to do is to select only annual data that matches the report year.  In which case it could maybe be called `SelectAnnualRecordsDurationXbrl` and have only `report_year` as a parameter, using that report year to construct the appropriate start/end dates?

Do you anticipate needing it to be able to select data where the report year is different from the start/end dates? Or non-annual frequencies? If we do want it to be more flexible, maybe just calling it `SelectDatesDurationXbrl` or `SelectSpecificDateRangesXbrl` would indicate that it has to be given exact matches? Do you need the list-of-ranges functionality now?

We'll need a similar kind of functionality for the instant tables. Could it be  generalized to work on either of them, by allowing either start/end date or just date?

Reply:
I wanted to enable the choosing of dates for specific report years bc it seems like sometimes the end-of-last-year data is important. I suppose we could do `SelectAnnualRecordsDurationXbrl` but it just seems much less flexible for other date selection scenarios that i'm sure will come up

the list of ranges for each report year is important bc... hopefully we'll add new years of data.

Reply:
I think maybe there aren't going to be a ton of different specific applications of this and we could just make them separate selecting transforms. See my comment below.